modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java

```diff
@@ -192,6 +193,12 @@ public void processOpts() {
             additionalProperties.put(PROVIDED_IN_ROOT, false);
         }
 
+        if (ngVersion.atLeast("9.0.0")) {                
+            additionalProperties.put(MODULE_WITH_PROVIDERS, true);
```

Comment:
Looks like the purpose of this flag is to drive adding a generic param to the ModuleWithProviders (as in angular 9 the default value for param (any) has been removed). It doesn't affect whether the module with providers is generated or not. Is that correct?
If so, perhaps a better name would be "ENFORCE_GENERIC_MODULE_WITH_PROVIDERS" or smth like that?
Or maybe even remove it from the java class, and just do conditionally in the template (based on ngVersion)?

Reply:
rename is fine. concerning the template condition, I think this is a bit difficult with the mustache template, so I think the current solution is fine.